### ISSUE TITLE
New notification providers

### ISSUE THREAD

**Issue by lokenx** [lokenx/plexrequests-meteor#164]
If anyone is interested in having notifications from providers other than Pushbullet and Pushover, please add them to this issue.

**New providers:**
 - [ ] Twitter
 - [ ] Email (Mailgun)

**Comment by duramato** [comment]
Twiiter would be cool to have ,either as DM or tweets

**Comment by lokenx** [comment]
Can you give an idea of how this would work? How would the app posting to your twitter timeline that someone has requested something help you? Should the users details be included? Direct Messages are an option but to whom are they directed?

**Comment by duramato** [comment]
It should make a tweet when someone requests something with his name and the show
For the DM you input your api key or whatever its needed and you specify the user you want to send it (much like sickrage implemetation) with the same info as the tweet

**Comment by lokenx** [comment]
But does it make this tweet from your own account or some other? Because then your followers would be reading these tweets?

And same for DM, from whom is the DM coming from? Can you DM yourself?

I'll see what SickRage does!

**Comment by duramato** [comment]
What i have on SickRage is : 
I made a dedicated account for the Tweets so that they dont come from my own acc and the DM also come from that one

**Comment by lokenx** [comment]
OK that makes sense, thanks!

**Comment by pickthenimp** [comment]
Prowl please!

**Comment by jrdnlc** [comment]
Boxcar support pleaseee
https://boxcar.io/developer

**Comment by wilsonic** [comment]
NotifyMyAndroid would be awesome.
http://www.notifymyandroid.com/
Even just simple email support would be useful.

**Comment by unixjeff** [comment]
I vote email, because then you can use ifttt to push it anywhere

**Comment by lokenx** [comment]
I like idea of email too. If I use Mailgun or similar, you'd just need to create an account to get an API key and such and that gets you 10,000 emails a month for free.

**Comment by jrdnlc** [comment]
email + iftt = awesomenesss

**Comment by unixjeff** [comment]
If you did email them you could do like a email subscription, so you don't have to create user accounts to receive notifications.

**Comment by lokenx** [comment]
@unixjeff not sure I get what you mean? Who would be receiving these emails?

@jrdnlc makes sense, when I look into this I'll see if can provide both options.

**Comment by unixjeff** [comment]
Like a subscription..  or distribution list so it would send the alerts to all emails subscribed

**Comment by unixjeff** [comment]
Newsletter was the word I was looking for  not subscription

**Comment by jrdnlc** [comment]
Here is an example of what i'm talking about @lokenx 
![screen shot 2015-11-17 at 6 37 25 pm](https://cloud.githubusercontent.com/assets/15741144/11231127/5099f9de-8d5a-11e5-8446-fd85f7845f79.png)

**Comment by lokenx** [comment]
This is the configuration variable that gets set for sending via SMTP, so as long as that works you'd be good to go!

```
smtp://USERNAME:PASSWORD@HOST:PORT/

**Comment by jrdnlc** [comment]
@lokenx So we are forced to use Mailgun?

**Comment by lokenx** [comment]
No, not necessarily. I should be able to provide both options, just depends if the SMTP config will work with whatever you have.

I'll also look into maybe a different email sending package that has more details, but right now I'm figuring out the twitter API.

**Comment by causefx** [comment]
What about Telegram?

https://core.telegram.org/api

PlexPy supports Telegram:
![image](https://cloud.githubusercontent.com/assets/16184466/11699878/dda24f90-9e83-11e5-95b3-43c77c504ead.png)


![image](https://cloud.githubusercontent.com/assets/16184466/11699829/8bbd42ac-9e83-11e5-9cd8-9797a8646cca.png)

**Comment by saitoh183** [comment]
+1 for telegram

**Comment by ThePierce** [comment]
Boxcar +1 as i use boxcar for other add-ons too. Works great!

**Comment by didyouexpectthat** [comment]
+10 for telegram.

**Comment by a5m0** [comment]
add e2e encryption to pushbullet notifications, seems like the only push service with e2e https://blog.pushbullet.com/2015/08/11/end-to-end-encryption/

**Comment by lokenx** [comment]
@a5m0 that wouldn't be possible for us to do, that would only work with official Pushbullet apps, unless I'm mistaking with the implementation.

**Comment by a5m0** [comment]
@lokenx it's just a symmetric key that you put on each side. So you would type it in on your phone client and put it into your sending client. (I don't think it would work on channels though)

**Comment by lokenx** [comment]
@a5m0 so only the phone would get it? What about web app? If you're interested in it, open a new ticket as it's not related to this particular one.

**Comment by a5m0** [comment]
Actually now after looking harder I'm not that sure if e2e encryption is available on push-to-self, but all their clients support it.

**Comment by rigrassm** [comment]
Going to work on adding IFTTT webhook support. While not a catch all solution, it opens up access to using just about any notification service if you're willing to do a little work.

**Comment by rigrassm** [comment]
Just created a new branch that incorporates basic IFTTT Maker support. Basically, this allows you to fire any IFTTT supported event whenever a request is made. Right now the data sent to IFTTT is statically set in the notifications section of plexrequests but the plan is to add in variable data to allow the user to customize the data being sent.

For information on the service and to create a free account, visit https://ifttt.com. Once you have an account, you can visit https://ifttt.com/maker to learn more about triggering events using the "Maker" channel.

**Comment by ajayjohn** [comment]
It would be handy to have Join supported (https://joaoapps.com/join/) It seems like one of the best options now since I'm sure you're aware, Pushbullet changed their business model and API calls are now limited.
Sickrage already supports Join too. Please look into this API.

**Comment by LuxAetern** [comment]
+1 for prowl

**Comment by jaretclifton** [comment]
+1 for basic SMTP mail notifications

**Comment by almontef** [comment]
+1 for basic SMTP mail notifications as well!

**Comment by zin105** [comment]
+1 for Join

**Comment by JagandeepBrar** [comment]
I have added support for Telegram, and created a pull request: #497.
